src/OrchardCore.Modules/OrchardCore.Autoroute/Indexes/AutoroutePartIndex.cs

```diff
@@ -92,7 +92,7 @@ public override async Task PublishedAsync(PublishContentContext context)
 
                 // Search for this part.
                 var contentTypeDefinition = _contentDefinitionManager.GetTypeDefinition(context.ContentItem.ContentType);
-                if (!contentTypeDefinition.Parts.Any(ctpd => ctpd.Name == nameof(AutoroutePart)))
+                if (contentTypeDefinition != null && !contentTypeDefinition.Parts.Any(ctpd => ctpd.Name == nameof(AutoroutePart)))
```

Comment:
This is an issue because then the AutoroutePart record will stay in the database and thus will keep an active route/url for that content item while it should dispose of the AutoroutePart record of that Content Item. But at the same time if the Content Definition has been removed then the user should not be able to make any change to that content item because it should fail to retrieve its content definition and warn about it beforehand.

Reply:
I don't recall exact steps that lead me to this case, maybe importing from migration. Regardless, I think we should be doing null check to prevent null reference exception.